akka-remote/src/main/scala/akka/remote/Remoting.scala

```diff
@@ -409,11 +411,9 @@ private[remote] class EndpointManager(conf: Config, log: LoggingAdapter) extends
   // Mapping between transports and the local addresses they listen to
   var transportMapping: Map[Address, AkkaProtocolTransport] = Map()
 
-  def retryGateEnabled = settings.RetryGateClosedFor > Duration.Zero
-  val pruneInterval: FiniteDuration = if (retryGateEnabled) settings.RetryGateClosedFor * 2 else Duration.Zero
-  val pruneTimerCancellable: Option[Cancellable] = if (retryGateEnabled)
-    Some(context.system.scheduler.schedule(pruneInterval, pruneInterval, self, Prune))
-  else None
+  val pruneInterval: FiniteDuration = settings.RetryGateClosedFor * 2
+  val pruneTimerCancellable: Cancellable =
+    context.system.scheduler.schedule(pruneInterval, pruneInterval, self, Prune)
```

Comment:
But that was not a documented feature, it was a bug. The docs are quite clear about that there is always a Gated state: http://doc.akka.io/docs/akka/2.3.14/scala/remoting.html#Lifecycle_and_Failure_Recovery_Model

Do you think we should keep it as it is, although this was not documented and contradicts the docs? Before 2.3 there was an explicit flag, then we removed the feature along with the corresponding tests and this remained somehow.

Not an easy question, maybe best to go conservative, log a Warning for the zero setting, and keep the current behavior.What about 2.4? Is it too late to fix it there?